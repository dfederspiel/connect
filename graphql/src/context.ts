import { PrismaClient } from '@prisma/client';
import { mockDeep } from 'jest-mock-extended';
import { DeepMockProxy } from 'jest-mock-extended/lib/mjs/Mock';

export type Context = {
  prisma: PrismaClient;
};

export type MockContext = {
  prisma: DeepMockProxy<PrismaClient>;
};

export const createMockContext = (): MockContext => {
  return {
    prisma: mockDeep<PrismaClient>(),
  };
};
