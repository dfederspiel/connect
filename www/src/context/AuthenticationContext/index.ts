export { useAuth, AuthProvider } from './AuthProvider';
export type { AuthContext, UserProfile } from './types';
