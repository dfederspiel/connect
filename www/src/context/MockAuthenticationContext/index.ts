export { MockAuthProvider } from './MockAuthProvider';
export { useProvideMockAuth } from './useProvideMockAuth';
