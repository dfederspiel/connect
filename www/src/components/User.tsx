import { Button, Typography } from '@mui/material';
import { useAuth } from '../context/AuthenticationContext';

const User = (): JSX.Element => {
  const auth = useAuth();

  return (
    <>
      <Typography>
        {auth.user}
        {!auth.user && (
          <Button
            onClick={() => {
              auth.signin();
            }}
          >
            Login
          </Button>
        )}
        {auth.user && (
          <Button
            onClick={() => {
              auth.signout();
            }}
          >
            Logout
          </Button>
        )}
      </Typography>
    </>
  );
};

export default User;
