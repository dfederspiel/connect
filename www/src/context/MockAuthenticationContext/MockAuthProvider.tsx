import { useProvideMockAuth } from '.';
import { AuthProvider } from '../AuthenticationContext/';
import { AuthContext } from '../AuthenticationContext/types';
// import { AuthModule, Mode } from "./AuthModule";
/**
 * AuthContext to be passed to any component that depends on user context.
 */
// const authContext = createContext<AuthContext>(null as any);

// Provider component that wraps your app and makes auth object ...
// ... available to any child component that calls useAuth().
// eslint-disable-next-line react/prop-types

interface MockAuthProviderProps {
  children?: JSX.Element;
  user?: string;
}

export const MockAuthProvider = ({
  children,
  user,
}: MockAuthProviderProps): JSX.Element => {
  const auth = useProvideMockAuth(user) as AuthContext;
  return <AuthProvider provider={auth}>{children}</AuthProvider>;
};
