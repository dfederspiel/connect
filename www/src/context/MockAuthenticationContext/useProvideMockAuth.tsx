import { AuthContext } from '../AuthenticationContext/types';

// eslint-disable-next-line @typescript-eslint/no-empty-interface
export interface User {}

export const useProvideMockAuth = (name?: string): AuthContext => {
  const user = name;

  const signin = async (userPolicyId?: string): Promise<void> => {
    if (userPolicyId) sessionStorage.setItem('currentPolicy', userPolicyId || '');
  };

  // eslint-disable-next-line @typescript-eslint/no-empty-function
  const signout = (): void => {};

  const token = async (): Promise<string | null> => '123';

  return {
    user,
    signin,
    token,
    signout,
  };
};
