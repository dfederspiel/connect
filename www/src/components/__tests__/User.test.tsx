import { act, fireEvent, render } from '@testing-library/react';
import { MockAuthProvider } from '../../context/MockAuthenticationContext';
import User from '../User';

describe('the user component', () => {
  describe('when the user is logged in', () => {
    const wrapper = ({ children }: any) => (
      <MockAuthProvider user="david@codefly.ninja">{children}</MockAuthProvider>
    );

    it('renders correctly with user context', () => {
      const { asFragment, getByRole } = render(<User />, { wrapper });
      expect(getByRole('button').textContent).toEqual('Logout');
      expect(asFragment()).toMatchSnapshot();
    });

    describe('and when the user clicks the logout button', () => {
      it('call the signout function', () => {
        const { getByRole } = render(<User />, { wrapper });
        act(() => {
          fireEvent.click(getByRole('button'));
        });
      });
    });
  });

  describe('when the user is not logged in', () => {
    const wrapper = ({ children }: any) => (
      <MockAuthProvider>{children}</MockAuthProvider>
    );
    it('renders correctly without user context', () => {
      const { asFragment, getByRole } = render(<User />, { wrapper });
      expect(getByRole('button').textContent).toEqual('Login');

      expect(asFragment()).toMatchSnapshot();
    });

    describe('and when the user clicks the login button', () => {
      it('call the signin function', () => {
        const { getByRole } = render(<User />, { wrapper });
        act(() => {
          fireEvent.click(getByRole('button'));
        });
      });
    });
  });
});
