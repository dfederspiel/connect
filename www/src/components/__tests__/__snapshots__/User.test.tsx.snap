// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the user component when the user is logged in renders correctly with user context 1`] = `
<DocumentFragment>
  <p
    class="MuiTypography-root MuiTypography-body1"
  >
    david@codefly.ninja
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Logout
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </p>
</DocumentFragment>
`;

exports[`the user component when the user is not logged in renders correctly without user context 1`] = `
<DocumentFragment>
  <p
    class="MuiTypography-root MuiTypography-body1"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Login
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </p>
</DocumentFragment>
`;
